src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,23 @@
+import Popup from "./Popup.js";
+
+class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleFormSubmit }) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  setSubmitAction(handleFormSubmit) {
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleFormSubmit();
+    });
+  }
+}
```

Comment:
Needs correction: you should also add a method for rendering loading, so you will be able to show/hide the loading state for the confirmation popup

Reply:
Did you just want the method added so that the option is available for future use? 

The figma design doesn't have a loading text on the popupWithConfirmation (delete card modal)